pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix

```diff
@@ -1,61 +1,43 @@
-{ stdenv, lib, fetchurl, makeDesktopItem, copyDesktopItems, makeWrapper,
-electron, libsecret }:
+{ lib
+, appimageTools
+, fetchurl
+}:
 
-stdenv.mkDerivation rec {
+let
   pname = "tutanota-desktop";
-  version = "3.119.3";
+  version = "3.119.10";
 
   src = fetchurl {
-    url = "https://github.com/tutao/tutanota/releases/download/tutanota-desktop-release-${version}/${pname}-${version}-unpacked-linux.tar.gz";
-    name = "tutanota-desktop-${version}.tar.gz";
-    hash = "sha256-TdjvU12nh1sTfGTdBn+7dbEunaF38YjDvceEns4iRbA=";
+    url = "https://github.com/tutao/tutanota/releases/download/tutanota-desktop-release-${version}/tutanota-desktop-linux.AppImage";
+    hash = "sha256-yxj23SmCzgLqGAAgy7XzQYXcICFNbDBZQr8DJIK+oPs=";
   };
 
-  nativeBuildInputs = [
-    copyDesktopItems
-    makeWrapper
-  ];
-
-  dontConfigure = true;
-  dontBuild = true;
-
-  desktopItems = makeDesktopItem {
-    name = pname;
-    exec = "tutanota-desktop";
-    icon = "tutanota-desktop";
-    comment = meta.description;
-    desktopName = "Tutanota Desktop";
-    genericName = "Email Reader";
+  appimageContents = appimageTools.extractType2 {
+    inherit pname version src;
   };
 
-  installPhase = ''
-    runHook preInstall
+in appimageTools.wrapType2 rec {
+  inherit pname version src;
 
-    mkdir -p $out/bin $out/opt/tutanota-desktop $out/share/tutanota-desktop
+  extraPkgs = pkgs: (appimageTools.defaultFhsEnvArgs.multiPkgs pkgs)
+    ++ [ pkgs.libsecret ];
 
-    cp -r ./ $out/opt/tutanota-desktop
-    mv $out/opt/tutanota-desktop/{locales,resources} $out/share/tutanota-desktop
+  extraInstallCommands = ''
+    mv $out/bin/${pname}-${version} $out/bin/${pname}
 
-    for icon_size in 64 512; do
-      icon=resources/icons/icon/$icon_size.png
-      path=$out/share/icons/hicolor/$icon_size'x'$icon_size/apps/tutanota-desktop.png
-      install -Dm644 $icon $path
-    done
+    install -Dm 444 ${appimageContents}/tutanota-desktop.desktop -t $out/share/applications
+    install -Dm 444 ${appimageContents}/tutanota-desktop.png $out/share/pixmaps
 
-    makeWrapper ${electron}/bin/electron \
```

Comment:
Check if Tutanota Desktop works fine with Wayland, if yes then keep makeWrapper and add `NIXOS_OZONE_WL` support

Reply:
I do not have Wayland to test this

Reply:
The way it is now works for my on Hyprland with XWayland. Using the args provided by @CactiChameleon9 it does not start for me. I think the relevant log line is this:

```
[57048:0118/104949.744896:ERROR:gpu_process_host.cc(957)] GPU process exited unexpectedly: exit_code=139
```

Using mesa with an AMD card btw (don't know how relevant that is).

Reply:
If you leave it in its current form (the one in this PR), it will continue to work for everyone, no? Save the wayland stuff for another PR.